src/odfesqlodbc/es_communication.cpp

```diff
@@ -471,7 +473,7 @@ int ESCommunication::ExecDirect(const char* query, const char* fetch_size_) {
         delete result;
         return -1;
     }
-    m_result_queue.push(std::unique_ptr< ESResult >(result));
+    m_result_queue.push(std::reference_wrapper< ESResult >(*result));
```

Comment:
The return value is lost here (SQL_SUCCESS or SQL_ERROR). Does it make sense to handle an error case here if SQL_ERROR is returned?